File: unida-jwt-server/src/main/ruby/me/dusanov/unida/jwt/simple_form_server.rb
#init auth
#    authProvider = createProvider().setNonces(new JsonArray().add("queiM3ayei1ahCheicupohphioveer0O"));

$vertx.create_http_server().request_handler() { |req|
  if (req.uri().==("/"))
    # Serve the index page
    req.response().send_file("index.html")
  elsif (req.uri().start_with?("/form"))
    req.response().set_chunked(true)
    req.set_expect_multipart(true)
    req.end_handler() { |v|
      req.form_attributes().names().each do |attr|
        req.response().write("Got attr #{attr} : #{req.form_attributes().get(attr)}\n")
      end
      req.response().end()
    }
    =begin
            //paste start
            UsernamePasswordCredentials credentials =
              new UsernamePasswordCredentials(req.formAttributes().get("username"), req.formAttributes().get("password"));

            authProvider.authenticate(credentials,user -> {
              //assertNotNull(user);
              System.out.println("=== got this:\n" + user);
            });
            //paste end
    =end
  else
    req.response().set_status_code(404).end()
  end
}.listen(8080)

File: unida-jwt-server/src/main/js/me/dusanov/unida/jwt/simple_form_server.js
//init auth
//    authProvider = createProvider().setNonces(new JsonArray().add("queiM3ayei1ahCheicupohphioveer0O"));

vertx.createHttpServer().requestHandler(function (req) {
  if (req.uri() == "/") {
    // Serve the index page
    req.response().sendFile("index.html");
  } else if (req.uri().startsWith("/form")) {
    req.response().setChunked(true);
    req.setExpectMultipart(true);
    req.endHandler(function (v) {
      Array.prototype.forEach.call(req.formAttributes().names(), function(attr) {
        req.response().write("Got attr " + attr + " : " + req.formAttributes().get(attr) + "\n");
      });
      req.response().end();
    });
    /*
            //paste start
            UsernamePasswordCredentials credentials =
              new UsernamePasswordCredentials(req.formAttributes().get("username"), req.formAttributes().get("password"));

            authProvider.authenticate(credentials,user -> {
              //assertNotNull(user);
              System.out.println("=== got this:\n" + user);
            });
            //paste end
    */
  } else {
    req.response().setStatusCode(404).end();
  }
}).listen(8080);
